ios/Exponent/Versioned/Core/Api/EXNotifications.m

```diff
@@ -92,41 +95,134 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId kernelServiceDeleg
+  UNMutableNotificationContent* content = [self _localNotificationFromPayload:payload];
 
   [EXUtil performSynchronouslyOnMainThread:^{
-    [RCTSharedApplication() presentLocalNotificationNow:notification];
+    UNNotificationRequest* request = [UNNotificationRequest requestWithIdentifier:content.userInfo[@"id"]
+                                                                          content:content
+                                                                          trigger:nil];
+    [[EXUserNotificationCenter sharedInstance] addNotificationRequest:request withCompletionHandler:^(NSError * _Nullable error) {
+      if (error != nil) {
+        reject(@"E_NOTIF", [NSString stringWithFormat:@"Could not add a notification request: %@", error.localizedDescription], error);
+      } else {
+        resolve(content.userInfo[@"id"]);
+      }
+    }];
   }];
 
-  resolve(notification.userInfo[@"id"]);
+}
+
+RCT_EXPORT_METHOD(addCategory: (NSString *)categoryId
+                  actions: (NSArray *)actions
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(__unused RCTPromiseRejectBlock)reject)
+{
+  categoryId = [self getScopedIdIfDetached:categoryId];
+
+  for (id action in actions) {
+    action[0] = [self getScopedIdIfDetached:action[0]];
+  }
+
+  NSMutableArray<UNNotificationAction *> * actionsArray = [[NSMutableArray alloc] init];
+  
+  for(NSArray *action in actions) {
+    /*
+     action[0] actionId
+     action[1] buttonTitle
+     action[2] flags
+     action[3] buttonTitle on keyboard
+     action[4] default text in textInput
+     */
+
+    EXCategoryAction * categoryAction = [EXCategoryAction parseFromArray:action];
+    [actionsArray addObject:[categoryAction getUNNotificationAction]];
+  }
+  
+  UNNotificationCategory * newCategory = [UNNotificationCategory categoryWithIdentifier:categoryId actions:actionsArray intentIdentifiers:@[] options:UNNotificationCategoryOptionNone];
+  
+  [[EXUserNotificationCenter sharedInstance] getNotificationCategoriesWithCompletionHandler:^(NSSet * categories) {
+    NSMutableSet * newCategories = [categories mutableCopy];
+    for( UNNotificationCategory * category in newCategories) {
+      if ([category.identifier isEqualToString:categoryId]) {
+        [newCategories removeObject:category];
+        break;
+      }
+    }
+    [newCategories addObject:newCategory];
+    [[EXUserNotificationCenter sharedInstance] setNotificationCategories:newCategories];
+    resolve(nil);
+  }];
 }
 
 RCT_EXPORT_METHOD(scheduleLocalNotification:(NSDictionary *)payload
                   withOptions:(NSDictionary *)options
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(__unused RCTPromiseRejectBlock)reject)
 {
-  UILocalNotification *notification = [self _localNotificationFromPayload:payload];
+  bool repeats = NO;
+  if (options[@"repeats"]) {
+    repeats = (BOOL)options[@"repeats"];
```

Comment:
`options[@"repeats"]` will be either `nil` or an `NSNumber`. To coalesce this value to a bool, use `boolValue` method on NSNumber